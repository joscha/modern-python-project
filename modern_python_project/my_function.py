def my_function(name: str = "World"):
    return f"Hello, {name}!"
