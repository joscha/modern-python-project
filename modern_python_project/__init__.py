from .my_function import my_function

__all__ = ["my_function"]
