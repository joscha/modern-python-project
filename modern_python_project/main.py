import click

from .my_function import my_function


@click.command()
@click.option("--name", prompt="Your name", help="The person to greet.")
def main(name):
    print(my_function(name=name))


if __name__ == "__main__":
    main()
