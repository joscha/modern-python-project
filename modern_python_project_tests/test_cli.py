from click.testing import CliRunner

from modern_python_project.main import main


def test_hello_world(snapshot):
    runner = CliRunner()
    result = runner.invoke(main, ["--name", "Peter"])
    assert result.exit_code == 0
    assert snapshot == result.output
