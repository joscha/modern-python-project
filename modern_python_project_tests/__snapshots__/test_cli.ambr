# serializer version: 1
# name: test_hello_world
  '''
  Hello, Peter!
  
  '''
# ---
